service/service.go

```diff
@@ -602,14 +604,34 @@ func (s *Service) VerifyToken(ctx context.Context, req *pbsvc.UserRequest) (*pbs
 		return nil, status.Error(codes.InvalidArgument, consts.ErrNilRequestIdentification.Error())
 	}
 
-	// verify token with database
+	// verify token against database
+	retrievedSecret, err := getMatchingToken(identity.GetToken())
```

Comment:
`getMatchingToken` the name is very misleading because it returns a secret. it is more like validating.